src/views/ApartmentApp/components/StudentApplication/components/HallSelection/components/HallListItem/index.js

```diff
@@ -25,6 +27,12 @@ const HallListItem = ({
   const [hallRankValue, setHallRankValue] = useState(1); // Rank drop-down menu value
```

Comment:
Does this make every hall item start with rank 1?

Reply:
Technically yes, but that is only as a fail-safe, to prevent the page from throwing an error before it has a chance to set the true value.
The JavaScript likes jump the gun with null errors if you accidentally forget to give things a valid initially placeholder values at moment the component is initialized

Reply:
Huh, okay!